lib/auth_app_web/live/user_live/settings.ex

```diff
@@ -11,6 +11,12 @@ defmodule AuthAppWeb.UserLive.Settings do
     </.header>
 
     <div class="space-y-12 divide-y">
+      <%!-- TODO: find a good way to style this --%>
+      <div :if={not @sudo_mode?} class="mt-8">
```

Comment:
I was thinking we would do this with a page redirect/on_mount callback, similar to GitHub. This also probably means we should have two timers? One for renewing the sudo (every 10 minutes) and another for accepting it (it is valid for 20 minutes).

Reply:
You say:

I try to access the settings page, on_mount checks if sudo mode is active. When the last auth is between 0 and 10 minutes ago, we proceed. Otherwise we redirect the user to login? Or do we accept anything between 0 and 20? But when would renewing happen?

Reply:
It is active for the last 20 minutes, requires renewal every 10. The idea is that the plug checks for 10 minutes, because the user may be on the page for a while, but we allow the controller and the other actions to be for 20 minutes. WDYT?